packages/api/src/beacon/routes/validator.ts

```diff
@@ -655,7 +656,19 @@ export function getReqSerializers(): ReqSerializers<Api, ReqTypes> {
       },
     },
 
-    publishAggregateAndProofs: reqOnlyBody(ArrayOf(ssz.phase0.SignedAggregateAndProof), Schema.ObjectArray),
+    publishAggregateAndProofs: {
+      writeReq: (signedAggregateAndProofs) => {
+        const fork = config.getForkName(signedAggregateAndProofs[0].message.aggregate.data.slot);
+        return {
+          body: ArrayOf(ssz.allForks[fork].SignedAggregateAndProof).toJson(signedAggregateAndProofs),
+          headers: {"eth-consensus-version": fork},
+        };
+      },
+      parseReq: ({body, headers}) => [
+        ArrayOf(ssz.allForks[headers["eth-consensus-version"]].SignedAggregateAndProof).fromJson(body),
```

Comment:
header must be lower case as Fastify will convert all headers to lower case, this is much better handled on the refactor branch but I think we are fine with skipping input validation here